### PR TITLE
Apply various ease-of-use changes that I developed in the past week.

### PR DESCRIPTION
**Pull request by theotherjimmy**
See the commit log for more details

### CODE REVIEW

**Comment by costanic** [comment]
I recommend targeting the 'dev' branch with this PR instead of 'master' since dev branch is the current working branch and should be merged into master at the next release.

File: Makefile [hunk 1]
```diff
@@ -34,7 +34,7 @@ all: Makefile .docker-image conf
 
 .PHONY: bash
 bash:
-	$(call docker_run, /bin/bash)
+	$(call docker_run, sh -c "${SOURCE_OE}; exec /bin/bash")
```
Comment (costanic) [inline Makefile]: where is SOURCE_OE defined?

File: Makefile [hunk 2]
```diff
@@ -32,6 +33,16 @@ all: Makefile .docker-image conf
 	cp ./update_default_resources.c ${POKY}/meta-pelion-edge/recipes-wigwag/mbed-edge-core/files/
 	$(call docker_run, make bb/${IMAGE_RECIPE})
 
+flash-%: all
+	$(foreach dev, $(wildcard /dev/${*}*),\
+		sudo umount ${dev} || true; \
+	)
+	sudo bmaptool copy \
```
Comment (costanic) [inline Makefile]: Is it possible to make this fall back to 'dd' if the user doesn't have bmaptool installed?

Comment (cmonr) [reply Makefile]: Shouldn't the tool be guaranteed to exist since this being run within a docker context?

Comment (costanic) [reply Makefile]: This recipe doesn't appear to be executed from a Docker context.  Check the "bitbake-%" and related "bb/%" recipes below for an example of recipes in this Makefile that run within Docker.

Comment (cmonr) [reply Makefile]: Ahh, I see. My mistake.

File: Makefile (hunk 1 above)
Comment (theotherjimmy) [reply Makefile]: That's missing from the commit. I'll add it back.

File: Makefile (hunk 2 above)
Comment (theotherjimmy) [reply Makefile]: Sure, we can have this fallback. I'll update this PR shortly.

**Comment by theotherjimmy** [comment]
> I recommend targeting the 'dev' branch with this PR instead of 'master' since dev branch is the current working branch and should be merged into master at the next release.

If this is the case, you should change the default branch so that PRs go there by default.

**Comment by costanic** [comment]
> > I recommend targeting the 'dev' branch with this PR instead of 'master' since dev branch is the current working branch and should be merged into master at the next release.
> 
> If this is the case, you should change the default branch so that PRs go there by default.

Yep, makes sense, and in fact that's the way our branches were configured up until very recently when a customer became confused and complained about it.  When the customer cloned the repo, the default branch (dev) was checked out, but dev wasn't stable at the moment and had some features that the customer didn't need yet.  Due to time zone differences, it took a few days to get the customer sorted out.

Anyway, the lesson learned was to make sure that master is always stable and every effort should be made to make sure that users check out stable code by default and without requiring the customer to be aware of our branching strategy, hence the switch back to 'master' being the default.  We can always port PRs from master to dev much more easily than dealing with a temporarily confused user.

**Comment by theotherjimmy** [comment]
That seems to imply that `dev` is unstable. That sounds like it would be very bad for development. Could we keep the development branch stable?

**Comment by costanic** [comment]
> That seems to imply that `dev` is unstable. That sounds like it would be very bad for development. Could we keep the development branch stable?

The only branch guaranteed to be stable is 'master'.  'dev' _should_ be stable, but it has not necessarily passed all regression tests at any moment in time.

**Comment by theotherjimmy** [comment]
>  has not necessarily passed all regression tests at any moment in time.

How/Why did we let that happen. It seems like something we should gate PRs on.

**Comment by costanic** [comment]
Not all tests are automated.  New features may not always be stable.

**Comment by theotherjimmy** [comment]
> Not all tests are automated. 

sounds like we have work do to here.


-----------------------------

Anyhow, I pushed fixes for your review comments and rebased onto `dev`

File: Makefile [hunk 3]
```diff
@@ -50,8 +51,8 @@ all: Makefile .docker-image conf
 	$(call docker_run, make bb/${IMAGE_RECIPE})
 
 .PHONY: bash
-bash: .docker-image
-	$(call docker_run, /bin/bash)
+bash:
```
Comment (costanic) [inline Makefile]: This looks like it might be an accidental change.  Why remove the dependency on .docker-image?

Comment (theotherjimmy) [reply Makefile]: That's a rebase mistake.

Comment (theotherjimmy) [reply Makefile]: The most recent force push should have fixed this mistake.

File: Makefile [hunk 4]
```diff
@@ -51,7 +52,7 @@ all: Makefile .docker-image conf
 
 .PHONY: bash
 bash: .docker-image
-	$(call docker_run, /bin/bash)
+	$(call docker_run, sh -c "${SOURCE_OE}; exec /bin/bash")
```
Comment (costanic) [inline Makefile]: This for me with the following error:

$ make bash
docker run --privileged=true --device /dev/loop0:/dev/loop0 --device /dev/loop1:/dev/loop1 --device /dev/loop2:/dev/loop2 --device /dev/loop3:/dev/loop3 --device /dev/loop4:/dev/loop4 --device /dev/loop5:/dev/loop5 --device /dev/loop6:/dev/loop6 --device /dev/loop7:/dev/loop7 -it -v /home/ncosta/workspace/build-dev/poky:/home/ncosta/poky -v /home/ncosta/.ssh:/home/ncosta/.ssh -v /home/ncosta/workspace/build-dev/build-env/Makefile:/home/ncosta/Makefile -e EDITOR=vim -e POKY=/home/ncosta/poky -e TEMPLATECONF=/home/ncosta/poky/meta-pelion-edge/conf --rm ncosta/ww-build-env:latest  sh -c "source /home/ncosta/workspace/build-dev/poky/oe-init-build-env /home/ncosta/workspace/build-dev/poky/build; exec /bin/bash"
sh: /home/ncosta/workspace/build-dev/poky/oe-init-build-env: No such file or directory
Makefile:72: recipe for target 'bash' failed
make: *** [bash] Error 1

Comment (theotherjimmy) [reply Makefile]: Oh, huh. That's probably an issue. Let's put off this change until another PR. I'll rebase it out.

**Comment by theotherjimmy** [comment]
@costanic I removed the modifications to the `bash` target.

**Pull request merged**